acceptancetests/assess_network_health.py

```diff
@@ -176,17 +184,214 @@ def connect_to_existing_model(self, client, target_model):
         if client.show_model().keys()[0] is not target_model:
             client.switch(target_model)
 
+    def assign_spaces(self, client):
+        """Assigns spaces to subnets
+        Currently this only supports EC2.
+        Name the spaces sequentially: space1, space2, space3, etc.
+        We require at least 3 spaces.
+
+        :param client: Juju client object with controller
+        """
+        if client.env.provider == "ec2":
+            log.info("Assigning network spaces on EC2")
+            subnets = yaml.safe_load(client.get_juju_output('list-subnets',
+                '--format=yaml'))
+            if not subnets:
+                raise Exception('No subnets defined in EC2')
+            idx = 0
+            for subnet in subnets['subnets'].keys():
+                # skip subnets with INFAN in the provider-id as they may be
+                # inherited from underlay and therefore we cannot assign a space
+                if 'INFAN' not in subnets['subnets'][subnet]['provider-id']:
+                    idx += 1
+                    client.juju('add-space', ('space{}'.format(idx), subnet))
+            if idx < 3:
+                raise Exception('3 subnets required for spaces assignment.')
+
+
+    def verify_spaces(self, client):
+        """Verify that spaces are set up proper and functioning
+        Currently this only supports EC2.
+
+        :param client: Juju client object with machines and spaces
+        """
+        if client.env.provider == "ec2":
+            spaces = yaml.safe_load(client.get_juju_output('list-spaces',
+                '--format=yaml'))['spaces']
+            # Remove fan subnets from spaces list to avoid confusion
+            for space in list(spaces):
```

Comment:
This probably is useful outside the method. Consider making a function for clarity and reuse.